### PR TITLE
Fix package name support for embedded composites

### PR DESCRIPTION
**Pull request by ratcashdev**
Enhances the test and fixes bugs with embedded composites and REF elements inside composites.

Related to https://github.com/real-logic/simple-binary-encoding/pull/904 and https://github.com/real-logic/simple-binary-encoding/issues/507

### CODE REVIEW

File: sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java [hunk 1]
```diff
@@ -60,6 +61,7 @@ public class JavaGenerator implements CodeGenerator
     private static final String PACKAGE_INFO = "package-info";
     private static final String BASE_INDENT = "";
     private static final String INDENT = "    ";
+    private static final Set<String> NO_PACKAGES = Collections.unmodifiableSet(new HashSet<>());
```
Comment (mjpt777) [inline sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java:64]: Why not EMPTY_SET. https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#emptySet()

Comment (ratcashdev) [reply sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java]: No idea, really. I use it all the time. Fixed.

File: sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java [hunk 2]
```diff
@@ -1276,10 +1284,11 @@ private void generateFixedFlyweightHeader(
         final String typeName,
         final String implementsString,
         final String buffer,
-        final String fqBuffer) throws IOException
+        final String fqBuffer,
+        final Set<String> additionalPackages) throws IOException
```
Comment (mjpt777) [inline sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java:1288]: `additionalPackages` -> `importedTypesPackages`?

File: sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java [hunk 3]
```diff
@@ -1608,6 +1644,25 @@ private CharSequence generateEnumLookupMethod(final List<Token> tokens, final St
...
     }
 
+    private StringBuilder generateImportStatements(final Set<String> packages, final String currentPackage)
+    {
+        final StringBuilder importsStatements = new StringBuilder();
+        for (final String candidatePackage : packages)
+        {
+            if (!candidatePackage.equalsIgnoreCase(currentPackage))
+            {
+                importsStatements.append("import ");
```
Comment (mjpt777) [inline sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java:1654]: append can be chained for a fluent API.

File: sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java (hunk 2 above)
Comment (ratcashdev) [reply sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java:1288]: as you wish, though I'd like to point out, that those the not the only packes imported inside `generateFileHeader` so I figured it better to name it additional...

Comment (ratcashdev) [reply sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java:1288]: no, only those that are identified by `scanPackagesToImport` and that do not match the current package name.

Comment (ratcashdev) [reply sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java]: renamed.

**Pull request merged**